### PR TITLE
Improve `getNamespacedName()` return type

### PR DESCRIPTION
**Pull request by staabm**
triggered by https://github.com/phpstan/phpstan-src/pull/3013/files#r1564724535

refs https://github.com/phpstan/phpstan-src/pull/3013

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -49,6 +49,10 @@
     "conflict": {
        "thecodingmachine/safe": "<1.1.3"
     },
+    "scripts": {
```
Comment (staabm) [inline composer.json:52]: I am not used to the psalm commands.. have to google them all the time.. added them here to ease contributing for non-psalm experienced people like me

Comment (Ocramius) [reply composer.json:52]: Not very happy to have `composer` as a wrapper to run scripts, but it's a non-blocker for this patch

File: src/SourceLocator/Ast/Strategy/NodeToReflection.php [hunk 2]
```diff
@@ -28,6 +29,10 @@ public function __invoke(
     ): ReflectionClass|ReflectionConstant|ReflectionFunction {
         $namespaceName = $namespace?->name?->name;
 
+        if ($namespaceName === '') {
+            throw new LogicException('Namespace name should never be empty');
+        }
```
Comment (Ocramius) [inline src/SourceLocator/Ast/Strategy/NodeToReflection.php:34]: Is this something we can test at all? Or is it an upstream type refinement required?

Comment (staabm) [reply src/SourceLocator/Ast/Strategy/NodeToReflection.php:34]: the upstream type is already refined, but the change itself is not yet released

https://github.com/nikic/PHP-Parser/pull/993

Comment (Ocramius) [reply src/SourceLocator/Ast/Strategy/NodeToReflection.php:34]: Let's then roll with this as-is: a lock file update will expose the type contradiction later on 👍

**Comment by Ocramius** [comment]
Thanks @staabm!

**Pull request merged**